builds/fabric8-che/plugins/keycloak-plugin-token-provider/src/main/java/com/redhat/che/keycloak/token/provider/util/KeycloakUserValidator.java

```diff
@@ -0,0 +1,236 @@
+            
+            LOG.debug("Openshift token = {}", openShiftToken);
+            String openShiftUser = getOpenShiftUser(openShiftToken);
+            if (isNullOrEmpty(openShiftUser)) {
+                return false;
+            }
+            LOG.debug("Openshift user = {}", openShiftUser);
+            return openShiftUser.equals(getOpenShiftProjectOwner());
+        }
+    }
+
+    /**
+     * Get the OpenShift token associated with a particular Keycloak token.
+     * @param keycloakToken the Keycloak token
+     * @return the OpenShift token
+     * @throws ServerException
+     * @throws UnauthorizedException
+     * @throws ForbiddenException
+     * @throws NotFoundException
+     * @throws ConflictException
+     * @throws BadRequestException
+     * @throws IOException
+     *
+     * @see TokenLoader
+     */
+    private String getOpenShiftToken(String keycloakToken) throws ServerException, UnauthorizedException,
+            ForbiddenException, NotFoundException, ConflictException, BadRequestException, IOException {
+        try {
+            // Ensure header has "Bearer:" prefix
+            String auth = keycloakToken.startsWith("Bearer") ? keycloakToken : "Bearer " + keycloakToken;
+            String openShiftToken = keycloakToOpenshiftTokenCache.get(auth);
+            return openShiftToken;
+        } catch (ExecutionException e) {
+            LOG.error("Exception while getting OpenShift token:", e);
+        }
+        return null;
+    }
+
+    /**
+     * Gets the OpenShift user associated with an OpenShift token.
+     * @param openShiftToken The OpenShift token
+     * @return
+     *
+     * @see UserLoader
+     */
+    private String getOpenShiftUser(String openShiftToken) {
+        try {
+            String currentUsername = openShiftTokenToUserCache.get(openShiftToken);
+            if (currentUsername.contains("@")) {
+                // Some usernames are email addresses
+                currentUsername = currentUsername.split("@")[0];
+            }
+            return currentUsername;
+        } catch (ExecutionException e) {
+            LOG.error("Exception while getting user:", e);
+        }
+        return null;
+    }
+
+    /**
+     * Get the owner of the current OpenShift namespace. Note that usernames may, in some cases,
+     * be email addresses.
+     * @return the username, as specified by OpenShift
+     */
+    private String getOpenShiftProjectOwner() {
+        try(OpenShiftClient client = new DefaultOpenShiftClient()) {
+            String namespace = client.getNamespace().split("-")[0];
+            return namespace;
+        }
+    }
+
+    /**
+     * Load openshift tokens, given keycloak tokens. If an error occurs, loads the
+     * empty string.
+     *
+     * @see CacheLoader
+     */
+    private class TokenLoader extends CacheLoader<String, String> {
+
+        @Override
+        public String load(String keycloakToken) {
+            String osoToken = null;
```

Comment:
initialize osoToken to "" this way you will not need to check for null before returning.